### PR TITLE
Add queryable support for TakeLast and SkipLast.

### PR DESCRIPTION
**Pull request by JonHanna**

### CODE REVIEW

**Comment by JonHanna** [comment]
Oops. Sorry, hang ten on this for a second...

**Comment by JonHanna** [comment]
Rectified. I hadn't checked my coverage and failed to realise the tests weren't failing because I wasn't actually triggering them. I'm afraid they don't pass, but do if I temporarily replace your bit with the naïve `Reverse().Skip().Reverse()` etc. so the failures they detect is in the Enumerable bit, so this can be merged into your working on that in the meantime.

**Comment by JonHanna** [comment]
Oops. This tries to merge into your master. Sorry.

File: src/System.Linq/tests/SkipLastTests.netcoreapp1.1.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+                Assert.Equal(default(T), expected.ElementAtOrDefault(-1));
+            }
+        }
+
+        public static IEnumerable<object[]> SkipLastData()
+        {
+            IEnumerable<int> counts = new[] { int.MinValue, -1, 0, 1, 2, 3, 4, 5, int.MaxValue };
+            IEnumerable<IEnumerable<int>> enumerables = from count in counts
+                                                        select Enumerable.Range(count, Math.Min(100, Math.Abs(count)));
```
Comment (jamesqo) [inline src/System.Linq/tests/SkipLastTests.netcoreapp1.1.cs]: Will fix later when I get the time

**Pull request closed without merging**